functional/kata-monitor/run.sh

```diff
@@ -0,0 +1,245 @@
+	[ $? -ne 0 ] && error_with_msg "cannot create container"
+	crictl start $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot start container $CONID"
+}
+
+stop_workload() {
+	crictl stop $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot stop container $CONID"
+	crictl rm $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot remove container $CONID"
+	crictl stopp $PODID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot stop pod $PODID"
+	crictl rmp $PODID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot remove pod $PODID"
+}
+
+is_sandbox_there() {
+	local podid=${1}
+	local sbs s
+
+	sbs=$(curl -s 127.0.0.1:8090/sandboxes)
+	[ $? -ne 0 ] && error_with_msg "cannot retrieve the pod list"
+	if [ -n "$sbs" ]; then
+		for s in $sbs; do
+			if [ "$s" = "$podid" ]; then
+				return $TRUE
+				break
+			fi
+		done
+	fi
+	return $FALSE
+}
+
+is_sandbox_there_iterate() {
+	local podid=${1}
+
+	for i in $(seq 1 $CACHE_UPD_TIMEOUT_SEC); do
+		is_sandbox_there "$podid" && return $TRUE
+		echo -n "."
+		sleep 1
+		continue
+	done
+
+	return $FALSE
+}
+
+is_sandbox_missing_iterate() {
+	local podid=${1}
+
+	for i in $(seq 1 $CACHE_UPD_TIMEOUT_SEC); do
+		is_sandbox_there "$podid" || return $TRUE
+		echo -n "."
+		sleep 1
+		continue
+	done
+
+	return $FALSE
+}
+
+###########################
+title "pre-checks"
+
+[ "$IAM" != "root" ] &&
+	error_with_msg "run this script as root please (user is \"${IAM}\")"
+
+crictl pods > /dev/null
+[ $? -ne 0 ] && error_with_msg "connection to the container engine failed"
+echo_ok "connection to the container engine ok"
+
+###########################
+title "start kata-monitor"
+
+[ ! -x "$KATA_MONITOR_BIN" ] && error_with_msg "kata-monitor binary not found"
+
+[ "$CRI_RUNTIME" = "crio" ] && args="--runtime-endpoint /run/crio/crio.sock"
+$KATA_MONITOR_BIN $args --log-level trace > "$MONITOR_LOG_FILE" 2>&1 &
+KATA_MONITOR_PID="$!"
+echo_ok "kata-monitor started"
+
+###########################
+title "create kata workloads"
+
+start_workload
+echo "Created POD (runc): $PODID"
+VANILLA_PODID="$PODID"
+VANILLA_CONID="$CONID"
+
+start_workload "$RUNTIME"
+echo "Created POD (kata): $PODID"
+echo_ok "PODs are running"
+
+###########################
+title "kata-monitor cache update checks"
+
+is_sandbox_there_iterate "$PODID" || error_with_msg "cache: missing pod $PODID"
+echo "kata pod $PODID found in kata-monitor cache"
+
+is_sandbox_there_iterate "$VANILLA_PODID" && error_with_msg "cache: got vanilla pod $VANILLA_PODID"
+echo "vanilla pod $VANILLA_PODID skipped from kata-monitor cache"
+echo_ok "cache updated"
+
+###########################
+title "kata-monitor metrics retrieval"
+
+curl -s 127.0.0.1:8090/metrics > "$METRICS_FILE"
+[ $? -ne 0 ] && error_with_msg "cannot retrieve metrics"
+
+FILESIZE=$(stat -c%s "$METRICS_FILE")
+echo "Metrics file ($FILESIZE bytes): ${METRICS_FILE}"
+[ $FILESIZE -le 90000 ] && error_with_msg "metrics file looks too short"
```

Comment:
Where does this value come from? Please can you create a variable for this and add a comment.

Reply:
Well, checking the file size doesn't make much sense as we also count the number of metrics for a kata pod: removing this check  and adding a check on the number of metrics retrieved.